### ISSUE TITLE
Click handler is behaving strangely 

### ISSUE THREAD

**Issue by novakw267** [ga-wdi-boston/game-project#474]
I am having an issue where the click handler is working, and my function to prevent overwrite is working. However on the first click no matter which box you click it is displaying on the visual board the letter, however not logging it in the console. This then doesn't register with my win function and allows the first box I select to be overwritten once, but after that it no longer is allowed to be over written. 

TLDR: first click isn't registering in the console. but it is on the visual board. 

https://github.com/novakw267/tic-tac-toe/blob/nuke-pave/assets/scripts/game.js

**Comment by bernardlee** [comment]
```javascript
const reset = function () {
  fullBoard = false;
  board = newBoard;
};
```

`board` and `newBoard` start out by pointing to two different arrays (reference types, this is important).  The statement `board = newBoard;` sets the variable `board` to reference the array that `newBoard` is referencing, not copy its contents.

The first click runs through `game()` and calls `yourMove()`, which modifies the array that `board` is referencing.  `reset()` gets called sometime afterward and sets `board` to reference the array that `newBoard` is referencing, which is an array of empty strings.  Once this occurs, there is no longer any identifier referencing the original array that `board` was referencing when the application first ran.

You're essentially starting with a new array after the first turn.  This is why your problem is occurring.

**Comment by novakw267** [comment]
That makes sense I will have to redo the reset function then in order to make it work. 

thank you!
